apps/designer/app/shared/design-system/components/primitives/dnd/handlers.stories.tsx

```diff
@@ -40,32 +41,59 @@ export const Playground = () => {
     },
   });
 
+  const autoScrollHandlers = useAutoScroll({ target: scrollRef });
+
   const dragProps = useDrag({
     onStart(event: any) {
-      // FIXME: can't have target here, won't work with iframe
       if (event.target.dataset.draggable === "false") {
         event.cancel();
         return;
       }
       dragItemRef.current = event.target;
+      autoScrollHandlers.setEnabled(true);
+    },
+    onMove: (poiterCoordinate: any) => {
+      dropTargetHandlers.handleMove(poiterCoordinate);
+      autoScrollHandlers.handleMove(poiterCoordinate);
     },
-    onMove: handleMove,
     onEnd() {
-      handleEnd();
+      dropTargetHandlers.handleEnd();
       setDropTargetRect(undefined);
       setPlacementIndicatorRect(undefined);
       handleHoldEnd();
+      autoScrollHandlers.setEnabled(false);
     },
     onShiftChange({ shifts }: any) {
-      console.log("shifts", shifts);
       if (dragItemRef.current) {
         dragItemRef.current.textContent = `shifted ${shifts}`;
       }
     },
   });
 
   return (
-    <Box {...dragProps} ref={rootRef} css={{ display: "flex" }}>
+    <Box
+      {...dragProps}
+      ref={(el) => {
+        dropTargetHandlers.rootRef(el);
+
+        // TODO: do same as above for consistency
```

Comment:
ref containers are bad for composition because you can't react to them as an effect, better to always pass a function like above, so it basically becomes reactive to ref chamges